proposals/pr002-hunkspace.md

```diff
@@ -0,0 +1,545 @@
+are out of scope.
+
+A CLDR-only data provider would be a data provider which fixes the
+"source" string to the value "cldr.unicode.org".
+
+## Requirements
+Some requirements distilled from the discussion above:
+
+1.  There must be a commonly understood data schema for each data set.
+     For example "CLDR unicode data",
+2.  It must be possible for that data to come from multiple sources.
+     E.g. "mozilla.org", vs "google.com", vs. "unicode.org".  Every
+    source that provides a specific data schema can have own data.
+3.  The key space is multi-dimensional. Each dimension is named, and
+    the set of dimensions is specific to each data schema.  Example
+    dimension name: `message_id`.
+4.  The dimensions of the key space are [not]{.c1} ordered.
+
+# Key encoding schema
+
+Taking into account the needs for the shape and form of the keys
+discussed above, and the requirements, here are proposed mappings.
+
+It is allowed for the key encoding to have different equivalent
+schemas, which can be used where convenient.  Two proposed schemas are
+below, one for URI, another for JSON.
+
+## URI
+
+The following mapping specifies how the dimensions of the "hunkspace"
+key are mapped to an URI.
+
+* Data schema (always required):
+  * Key: `scheme`
+  * URI: "scheme"
+  * Remark: Perhaps we could sync up the namings so that we don't need to
+	maintain an active mapping like this.  Every scheme must define the data
+	schema for each value under a specific key. The data user is supposed to be
+	familiar with the data schema and able to parse the returned conforming
+	data.
+* Source (always required):
+  * Key: `source`
+  * URI: "authority"
+  * Remark:  We propose here the authority should be using the FQDN form such
+    as `icu4x.unicode.org`, as this may be more immediately recognizable due
+	to its prevalent use on the Internet.
+* Dimension
+  * Key: (a simple string)
+  * URI: path component such as `...locale_id/42/...`
+  * Remark: Every path component becomes a pair of path elements separated by
+  * a forward slash (`/`).
+
+For a multi-dimensional key, there may be several equivalent URI
+mappings, corresponding to different possible orderings of the
+dimensions and their values.  This allows us to omit any of the
+dimensions when specifying the URI, which can prove to be useful in
+query formulation.   It eliminates the guesswork, as every URI can be
+readily parsed into a key, and eliminates the need to introduce a
+hierarchical ordering in the URI, if such a hierarchy does not need to
+exist.
+
+> It is unclear at the moment if any of the encodings should be the
+"canonical" one.
+
+> We may need to either keep all dimensions flat.
+
+Example encodings are given below.
+
+```
+Key: {
+  source = "fuchsia.dev",
+  app="weirdy-birds",
+  message_id=42,
+  locale="en-US"
+}
+Value: "Hello, {user}"
+```
+
+### Example 1: Unicode data
+
+Always required dimensions are prefixed with "@" in the key.  Implementations
```

Comment:
Unless using the `@` is standard, I personally am not a fan of this. Perhaps this could be handled on the backend and be standardized? Or are there specific cases where the encoding keys will have different requirements?

Reply:
I don't think it is a standard in any way, perhaps the answer here is that we simply use flat key names, and use a convention that these fields are required.